ykrt/src/compile/jitc_yk/codegen/mod.rs

```diff
@@ -24,6 +24,7 @@ pub(crate) trait CodeGen: Send + Sync {
         m: Module,
         mt: Arc<MT>,
         hl: Arc<Mutex<HotLocation>>,
+        sp_offset: Option<usize>,
```

Comment:
What is `sp_offset`?

Reply:
There's a doc string on the implementation of codegen. Maybe we should move all of the doc strings over to the trait instead?

Reply:
I'm not going to pretend I understand that sentence :p

Reply:
The docstring is here: https://github.com/ykjit/yk/pull/1422/files#diff-2d8235b02f7af5cbc0cef1cf9b9789aaf37ee8bee882f4bd0f9c266e1d142edbR61

I feel like it's more relevant there. But I can copy it over if you insist.

Reply:
You can reference it with rustdoc syntax, that way it can't get out of sync.

Reply:
And there's another here: https://github.com/ykjit/yk/pull/1422/files#diff-b7513e8c31bfe56ee2b41fa041e8632d81f4bafc2e759b22b597f3867359269eR156

Reply:
Right, but when a dummy like me reads local code, they won't know this global context. rustdoc links (which editors often understand) are one solution to this.

Reply:
The problem with this design is that we are often passing values through various files. Do we always copy the docstrings to every single occurence? I don't know what a good standard for that is. e.g. do we also comment `m, mt, hl`?

Reply:
For example. Should we also comment `sp_offset` here:
- https://github.com/ykjit/yk/pull/1422/files#diff-b7513e8c31bfe56ee2b41fa041e8632d81f4bafc2e759b22b597f3867359269eR124
- https://github.com/ykjit/yk/pull/1422/files#diff-b7513e8c31bfe56ee2b41fa041e8632d81f4bafc2e759b22b597f3867359269eR162

The second of which seems even more important than the trait you commented on which we are hardly going to look at very often.

Reply:
@ltratt I've tried to improve the `sp_offset` doc strings. Sadly, Rust doesn't allow us to use doc strings for function parameters so I've used normal comments instead. But I've used doc strings wherever possible: [6589d14](https://github.com/ykjit/yk/pull/1422/commits/6589d14b0dfb171bd513fba0aebd78c3241476e4)